Makefile

```diff
@@ -9,6 +9,14 @@ CC = gcc
 AS = $(CC)
 LD = $(CC)
 RM = rm -f
+SYSTEM=$(shell uname 2>/dev/null)
+ifeq ($(SYSTEM), Darwin)
+SHARED_OBJ = libbcrypt.$(VERSION).dylib
+SHAREDFLAGS = -shared -W1,-soname,$(SHARED_OBJ)
+else
+SHARED_OBJ = libbcrypt.$(VERSION).so
+SHAREDFLAGS = -shared -Wl,-soname,$(SHARED_OBJ)
```

Comment:
There isn't -soname option with darwin, maybe.
You can use -install_name instead of -soname.